_posts/2023-05-17-otp-26-highlights.md

```diff
@@ -0,0 +1,902 @@
+ok
+```
+
+It is also possible to use a [custom ordering
+fun](https://www.erlang.org/doc/man/io.html#format-1).  For example,
+to order the map elements in reverse order based on their keys:
+
+```erlang
+4> io:format("~Kp\n", [fun(A, B) -> A > B end, M]).
+#{c => 3,b => 2,a => 1}
+ok
+```
+
+There is also a new
+[maps:iterator/2](https://www.erlang.org/doc/man/maps.html#iterator-2)
+function that supports iterating over the elements of the map in a more
+intutive order. Examples will be shown in the next section.
+
+## Map comprehensions
+
+In OTP 25 and earlier, it was common to combine `maps:from_list/1` and
+`maps:to_list/1` with list comprehensions. For example:
+
+```erlang
+1> M = maps:from_list([{I,I*I} || I <- lists:seq(1, 5)]).
+#{1 => 1,2 => 4,3 => 9,4 => 16,5 => 25}
+```
+
+In OTP 26, that can be written more succinctly with a [**map comprehension**](https://www.erlang.org/doc/reference_manual/expressions.html##comprehensions):
+
+```erlang
+1> M = #{I => I*I || I <- lists:seq(1, 5)}.
+#{1 => 1,2 => 4,3 => 9,4 => 16,5 => 25}
+```
+
+With a **map generator**, a comprehension can now iterate over the
+elements of a map. For example:
+
+```erlang
+2> [K || K := V <- M, V < 10].
+[1,2,3]
+```
+
+Using a map comprehension with a map generator, here is an example
+showing how keys and values can be swapped:
+
+```erlang
+3> #{V => K || K := V <- M}.
+#{1 => 1,4 => 2,9 => 3,16 => 4,25 => 5}
+```
+
+Map generators accept map iterators as well as maps. Especially useful
+are the ordered iterators returned from the new
+[maps:iterator/2](https://www.erlang.org/doc/man/maps.html#iterator-2)
+function:
+
+```erlang
+4> AM = #{a => 1, b => 2, c => 1}.
+#{c => 1,a => 1,b => 2}
+5> [{K,V} || K := V <- maps:iterator(AM, ordered)].
+[{a,1},{b,2},{c,1}]
+6> [{K,V} || K := V <- maps:iterator(AM, reversed)].
+[{c,1},{b,2},{a,1}]
+7> [{K,V} || K := V <- maps:iterator(AM, fun(A, B) -> A > B end)].
+[{c,1},{b,2},{a,1}]
+
+```
+
+Map comprehensions were first suggested in [EEP 58](https://www.erlang.org/eeps/eep-0058).
+
+## Inlined `maps:get/3`
+
+In OTP 26, the compiler will inline calls to
+[maps:get/3](https://www.erlang.org/doc/man/maps.html#get-3), making them slightly
+more efficient.
+
+## Improved `maps:merge/2`
+
+When merging two maps, the
+[maps:merge/2](https://www.erlang.org/doc/man/maps.html#merge-2)
+function will now try to reuse the [key
+tuple](https://www.erlang.org/doc/efficiency_guide/maps.html#how-small-maps-are-implemented)
+from one of the maps in order to reduce the memory usage for maps.
+
+For example:
+
+```erlang
+1> maps:merge(#{x => 13, y => 99, z => 100}, #{x => 0, z => -7}).
+#{y => 99,x => 0,z => -7}
+```
+
+The resulting map has the same three keys as the first map, so it can reuse the
+key tuple from the first map.
+
+This optimization is not possible if one of the maps has any key not present
+in the other map. For example:
+
+```erlang
+2> maps:merge(#{x => 1000}, #{y => 2000}).
+#{y => 2000,x => 1000}
```

Comment:
There's a similar optimisation done for the `=>` operator in updates that is new - before it was only done for the `:=` updates. Is this worth mentioning?

Reply:
Yes, it is. I forgot about this optimization.